src/Octoshift/GithubApi.cs

```diff
@@ -38,22 +38,30 @@ public virtual async Task<string> CreateTeam(string org, string teamName)
 
         public virtual async Task<IEnumerable<string>> GetTeamMembers(string org, string teamName)
         {
-            var url = $"https://api.github.com/orgs/{org}/teams/{teamName}/members";
+            var url = $"https://api.github.com/orgs/{org}/teams/{teamName}/members?per_page=100";
 
-            var response = await _client.GetAsync(url);
-            var data = JArray.Parse(response);
+            var teamMembers = new List<string>();
+            await foreach (var response in _client.GetAllAsync(url))
```

Comment:
what do you think about doing the foreach in GithubClient and just returning a JArray to GithubApi? Would move a bunch of repetitive code out of Api and into Client.

That's what I did in AdoClient.

Reply:
If I do the `foreach` in the client it takes away the flexibility of using the async stream. This way the client just doesn't care how to iterate over the data it just hands over the iterator and in the Api (which is kinda our business) we can decide how to iterate. It happens that in case of `GetTeamMembers` or `GetRepos` we actually want everything but even in those cases we might want to get data in parallel so we can use `Parallel.ForEachAsync`. Or  we might want to get the first 10 pages, etc. 
Now I realize that it might never happen in the CLI but even if I return `JArray` it doesn't save us a lot of lines since we still need to iterate over the data and parse the `JArray` with a `Select`. Here we can also use `Select` if we import the `System.Linq.Async` package 😃

Reply:
I'd much rather see that complexity pushed down into GithubClient.

If we really do need the flexibility I'd rather have 2 overloads in GithubClient, but given that today we'd only ever use the one that returns a JArray (and I don't see that changing anytime soon), IMO it makes more sense to push this down into the Client so we don't need to write the same few lines of code every time we call something with paging.

Reply:
I'd much rather see that complexity pushed down into GithubClient.

If we really do need the flexibility I'd rather have 2 overloads in GithubClient, but given that today we'd only ever use the one that returns a JArray (and I don't see that changing anytime soon), IMO it makes more sense to push this down into the Client so we don't need to write the same few lines of code every time we call something with paging.

Reply:
then in the Api class we can just do something like:

```
return await _client.GetAllAsync(url).Select(x => (string)x["login"]);

Reply:
addressed in 15bb7df1d55aaaad17ce713069b427e43b92d04b